includes/Pages/PageRequestFormManagement.php

```diff
@@ -0,0 +1,313 @@
+                    'email' => $form->getEmailHelp(),
+                    'comment' => $form->getCommentHelp(),
+                ]);
+
+                $this->assign('createMode', true);
+                $this->setTemplate('form-management/edit.tpl');
+
+                return;
+            }
+
+            $proceed = true;
+
+            if (RequestForm::getByPublicEndpoint($database, $form->getPublicEndpoint(), $domainId) !== false) {
+                SessionAlert::error("The chosen public endpoint is already in use. Please choose another.");
+                $proceed = false;
+            }
+
+            if (preg_match('/^[A-Za-z][a-zA-Z0-9-]*$/', $form->getPublicEndpoint()) !== 1) {
+                SessionAlert::error("The chosen public endpoint contains invalid characters");
+                $proceed = false;
+            }
+
+            if (RequestForm::getByName($database, $form->getName(), $domainId) !== false) {
+                SessionAlert::error("The chosen name is already in use. Please choose another.");
+                $proceed = false;
+            }
+
+            if ($form->getOverrideQueue() !== null) {
+                /** @var RequestQueue|bool $queue */
+                $queue = RequestQueue::getById($form->getOverrideQueue(), $database);
+                if ($queue === false || $queue->getDomain() !== $domainId || !$queue->isEnabled()) {
+                    SessionAlert::error("The chosen queue does not exist or is disabled.");
+                    $proceed = false;
+                }
+            }
+
+            if ($proceed) {
+                $form->save();
+                Logger::requestFormCreated($database, $form);
+                $this->redirect('requestFormManagement');
+            }
+            else {
+                $this->populateFromObject($form);
+                WebRequest::setSessionContext('preview', [
+                    'main' => $form->getFormContent(),
+                    'username' => $form->getUsernameHelp(),
+                    'email' => $form->getEmailHelp(),
+                    'comment' => $form->getCommentHelp(),
+                ]);
+
+                $this->assign('createMode', true);
+                $this->setTemplate('form-management/edit.tpl');
+            }
+        }
+        else {
+            $this->populateFromObject(new RequestForm());
+            WebRequest::setSessionContext('preview', null);
+            $this->assign('hidePreview', true);
+
+            $this->assignCSRFToken();
+            $this->assign('createMode', true);
+            $this->setTemplate('form-management/edit.tpl');
+        }
+    }
+
+    protected function view()
+    {
+        $database = $this->getDatabase();
+
+        /** @var RequestForm $form */
+        $form = RequestForm::getById(WebRequest::getInt('form'), $database);
+
+        if ($form->getDomain() !== Domain::getCurrent($database)->getId()) {
+            throw new AccessDeniedException();
```

Comment:
There's missing constructor parameters on the AccessDeniedException here and further down on L195; will fix later.